server/src/main/java/org/opensearch/cluster/routing/OperationRouting.java

```diff
@@ -116,6 +122,20 @@ public boolean ignoreAwarenessAttributes() {
         return this.awarenessAttributes.isEmpty() || this.ignoreAwarenessAttr;
     }
 
+    @Inject
+    public void setClusterService(ClusterService clusterService) {
+        this.clusterService = clusterService;
+    }
+
+    @Inject
+    public void setWrrShardsCache(WRRShardsCache wrrShardsCache) {
+        this.wrrShardsCache = wrrShardsCache;
+    }
```

Comment:
Injecting it in the constructor would require number of unit tests to be changed where this constructor is getting called. And this would require a change in ClusterService where we are constructing OperationRouting obj. Which mean we require to inject WRRShardCache in ClusterService as well as modify all the tests where OperationRouting is constructed. Wanted to reduce the number of changes so did it this way.